aten/src/ATen/native/native_functions.yaml

```diff
@@ -3518,6 +3518,15 @@
   variants: method
   named_guard: False
 
+- func: result_type(Tensor tensor, Tensor other) -> ScalarType
```

Comment:
how did you decide on these overloads?  For example, I'm a bit surprised this doesn't follow the `pow` pattern.

Reply:
My justification for this was based on `eq`, which supports `torch.eq(tensor, tensor)` or `torch.eq(tensor, scalar)` but not `torch.eq(scalar, tensor)`.

It's not ideal. I could add all possible combinations of overloads, but that seems messy. I assume there's a better way of doing this since it works both ways for `add` but I haven't investigated how that works. Would you be OK with extending this to support `(Scalar, Tensor)` and `(Scalar, Scalar)` in a separate PR?

Reply:
ya, separate PR sounds fine.  It also seems fine to not add this, assuming pow doesn't do type promotion (otherwise, you couldn't use this function to test pow).

Reply:
added all combinations (including scalar scalar which pow doesn't have because add does).